### PR TITLE
Adding Collection/Delivery info to entries and Photos from Google Drive

### PR DESCRIPTION
**Pull request by Johanna-hub**
- Adds whether a business does collection, delivery or both to PlaceItem
- Pulls in photos from Google Drive to use as image source
- Fixes Tags display in PlaceItem
- Links business name and photo to the single entry page

### CODE REVIEW

File: components/app/PlaceItem.jsx [hunk 1]
```diff
@@ -5,11 +5,12 @@ import { isLast } from '../utils';
...
 import { Box, Text, Image } from '../atoms';
 import { Row } from '../molecules';
-
+import { Link } from "gatsby";
+import _ from "lodash";
 
 
 const PlaceImage = ({ source, ...props }) => (
-  <Image source={source} {...props} />
+  <Image source={"http://drive.google.com/uc?export=view&id=" + source.split("=")[1]} {...props} />
```
Comment (macintoshhelper) [inline components/app/PlaceItem.jsx]: This kind of logic should exist in the container component (`PlaceItemContainer`).

I would probably do something like `if (source.startsWith('drive:')) { src = source.replace('drive:', 'http://drive.google.com/uc?export=view&id=')...`, to make sure normal URLs still work

File: components/app/PlaceItem.jsx [hunk 2]
```diff
@@ -57,26 +58,28 @@ const Tag = ({ type, ...props }) => (
...
       </CategoryName>
       <CollectionMethod>
-        {(category || '').toUpperCase()}
+        {(delivery === "TRUE" ? "Delivery" : '')}
+        {(collection === "TRUE" && delivery === "TRUE" ? " & Collection" : collection === "TRUE" ? "Collection" : '')}
       </CollectionMethod>
     </Row>
     <PlaceName mb={2}>
-      {name}
+    <Link to={`/${_.kebabCase(name)}`} style={{ textDecoration: 'none' }}>{name}</Link>
```
Comment (macintoshhelper) [inline components/app/PlaceItem.jsx]: I would probably use something like `to={`/business/${_kebabCase(name)}` (need to match whatever is in `gatsby-node.js` though)

File: components/app/PlaceItem.jsx [hunk 3]
```diff
@@ -57,26 +58,28 @@ const Tag = ({ type, ...props }) => (
...
     </PlaceName>
-    {tags && (
+    <TagText> {tags && (
```
Comment (macintoshhelper) [inline components/app/PlaceItem.jsx]: This `<TagText>` will break native platforms, e.g. Figma, etc. There'll be `<div>`s/`<View>`s inside of a `<span>`/`<p>`/`<Text>` element

File: components/app/PlaceItem.jsx [hunk 4]
```diff
@@ -85,11 +88,13 @@ const PlaceItemContainer = ({ children, ...props }) => (
     {children}
   </Box>
 );
-const PlaceItem = ({ children, place: { name, category, source, tags }, ...props }) => (
+const PlaceItem = ({ children, place: { name, category, source, tags, collection, delivery }, ...props }) => (
   <PlaceItemContainer {...props}>
+  <Link to={`/${_.kebabCase(name)}`}>
```
Comment (macintoshhelper) [inline components/app/PlaceItem.jsx]: `components` should be used for universal React code, so platform/framework specific imports like `<Link>` will break the `Figma` export, React Native, etc. I would create a new `Link` component with something like this:

```js
const Link = Platform.select({
  web: require('gatsby').Link,
  figma: React.Fragment
})
```

I understand that this adds extra friction/confusion :/ , but by doing this, this adds future proofing for having a React Native app in the future, or using the Figma component exports.

File: src/pages/index.js [hunk 5]
```diff
@@ -63,6 +68,9 @@ export const query = graphql`
           Category_1
           Category_2__only_if_relevant_
           Tags
+          Collection
```
Comment (macintoshhelper) [inline src/pages/index.js:81]: 👍

File: src/pages/index.js [hunk 6]
```diff
@@ -22,19 +22,24 @@ const categories = [{
...
+    Delivery,
+    Image_Url,
   } = {},
 }) => ({
   name: Name,
   category: Category_1,
-  source: 'https://media-cdn.tripadvisor.com/media/photo-s/15/7d/ca/8a/bakery.jpg',
-  tags: normaliseTag(Tags) || ['Fruit & Veg', 'Bread'],
+  source: Image_Url,
+  tags: Tags.split(","),
```
Comment (macintoshhelper) [inline src/pages/index.js]: This should be a pure function I think, e.g. `normaliseTag` would do `tags.split`, etc.

Implemented in 4c762c0

File: components/app/PlaceItem.jsx (hunk 1 above)
Comment (Johanna-hub) [reply components/app/PlaceItem.jsx]: I'm not really sure how to do this!  I've had a go but am a bit clueless where it goes in `(PlaceItemContainer)`.  Do you mind adding it sorry!

File: components/app/PlaceItem.jsx (hunk 2 above)
Comment (Johanna-hub) [reply components/app/PlaceItem.jsx]: The way it is currently set up it is http://localhost:8000/ashbys - there is no /business before it.  Do you mean we should add it?

File: components/app/PlaceItem.jsx (hunk 4 above)
Comment (Johanna-hub) [reply components/app/PlaceItem.jsx]: ah ok, so we shouldn't do Link because it is Gatsby specific?

File: components/app/PlaceItem.jsx (hunk 2 above)
Comment (macintoshhelper) [reply components/app/PlaceItem.jsx]: Yes, that's my suggestion. Up to you though. My idea is that it future proofs against a business happening to be named `/about` and conflicting with the actual `/about` page. It's more of something to consider for user-generated content/accounts though. Should be fine to stick to `/ashbys` if:

- business names are being checked manually
- there's a duplicate business name to id conversion process (e.g. two Gail's become `gails`, `gails-2`)
- there's a name blacklist/renaming process (business `About` gets renamed to `about-brixton`).

File: components/app/PlaceItem.jsx (hunk 4 above)
Comment (macintoshhelper) [reply components/app/PlaceItem.jsx]: I'll commit an example. The idea is to use components that work cross-platform, or to create a new cross-platform component, to support `react-native-web`, `react-figma`, `react-sketchapp`, etc.

example using platform filename resolving (when you import, the relevant platform file is imported):

`components/Link/index.web.js`

```js
export { Link as default } from 'gatsby';
```

`components/Link/index.figma.js`

```js
export { Fragment as default } from 'react';
// This acts as a stub (mock component that doesn't do anything, but carries on rendering its children)
```

Then you can do `import Link from '../Link';`.

I hope this makes sense. This isn't a common practice, but is necessary for the `react-figma` app to work, but the goal is that most of this work should be abstracted away, for example by releasing a `react-link-primitives` package to `npm`, which can be imported into any React project: Gatsby, React Native, React Figma, etc 🙂.

File: components/app/PlaceItem.jsx (hunk 1 above)
Comment (macintoshhelper) [reply components/app/PlaceItem.jsx]: Added this to 55f8459 and edb87f0 . The idea is that state "mutations" like this are more visible in the parent/container component.

File: components/app/PlaceItem.jsx (hunk 4 above)
Comment (macintoshhelper) [reply components/app/PlaceItem.jsx]: Implemented/documented in b1c1956 & 341abcb .

File: components/app/PlaceItem.jsx (hunk 2 above)
Comment (jamiecoe) [reply components/app/PlaceItem.jsx]: Good points! In 6adc23f, I've updated the business routes
 now to be `/business/${_kebabCase(name)`. I've also thrown an error if two businesses have the same name so the build will be blocked and someone will need to correct the spreadsheet. In the future we can add a duplicate business name to id conversion process - I guess the priority now is getting the site up and running

Comment (Johanna-hub) [reply components/app/PlaceItem.jsx]: Thanks so much @jamiecoe.  Yes, def just need to get something up there now!

**Pull request merged**